integr8ly/integreatly-operator
# MGDAPI-4923 - generalise index building script

# Issue link
[MGDAPI-4923](https://issues.redhat.com/browse/MGDAPI-4923)

# What
Generalise the index building script and allow it to work for all 3 products (3scale/rhsso/mcg)

# Verification steps
Ensure that you can successfully build index images for all three products. You can verify the bundle chain in the index images using the script from @MStokluska

```
get_bundle_chain(){
        INDEX_IMAGE=${1}
        DOCKER_IMAGE=$(docker run -d -p 50051:50051 $INDEX_IMAGE)
        grpcurl -plaintext localhost:50051 api.Registry/ListBundles | jq '{packageName: .packageName, name: .csvName, version: .version, bundlePath: .bundlePath, replaces: .replaces, skipRange: .skipRange, channelName: .channelName, dependencies: .dependencies}'
        docker kill $DOCKER_IMAGE
}
```
